commandline/src/test/java/tech/pegasys/eth2signer/commandline/valueprovider/YamlConfigFileDefaultProviderTest.java

```diff
@@ -0,0 +1,119 @@
+    final String[] args = cmdArgs.split(" ");
+    commandLine.parseArgs(args);
+
+    assertThat(eth2SignerCommand.getHttpListenPort()).isEqualTo(6001);
+    assertThat(eth2SignerCommand.getKeyConfigPath()).isEqualTo(Path.of("./keys_yaml"));
+  }
+
+  @Test
+  void unknownOptionsInConfigFileThrowsExceptionDuringParsing(@TempDir final Path tempDir)
+      throws IOException {
+    final String extraYamlOptions = CmdlineHelpers.validBaseYamlOptions() + "extra-option: True\n";
+    final File configFile =
+        Files.writeString(tempDir.resolve("config.yaml"), extraYamlOptions).toFile();
+    final Eth2SignerCommand eth2SignerCommand = new Eth2SignerCommand();
+    final CommandLine commandLine = new CommandLine(eth2SignerCommand);
+    commandLine.registerConverter(Level.class, Level::valueOf);
+    commandLine.setDefaultValueProvider(new YamlConfigFileDefaultProvider(commandLine, configFile));
+
+    final String[] args = CmdlineHelpers.validBaseCommandOptions().split(" ");
+    assertThatExceptionOfType(CommandLine.ParameterException.class)
+        .isThrownBy(() -> commandLine.parseArgs(args))
+        .withMessage("Unknown option in yaml configuration file: extra-option");
+  }
+
+  @Test
+  void invalidYamlConfigFileThrowsExceptionDuringParsing(@TempDir final Path tempDir)
+      throws IOException {
+    final String extraYamlOptions = CmdlineHelpers.validBaseYamlOptions() + "extra-option= True\n";
+    final File configFile =
+        Files.writeString(tempDir.resolve("config.yaml"), extraYamlOptions).toFile();
+    final Eth2SignerCommand eth2SignerCommand = new Eth2SignerCommand();
+    final CommandLine commandLine = new CommandLine(eth2SignerCommand);
+    commandLine.registerConverter(Level.class, Level::valueOf);
+    commandLine.setDefaultValueProvider(new YamlConfigFileDefaultProvider(commandLine, configFile));
+
+    final String[] args = CmdlineHelpers.validBaseCommandOptions().split(" ");
+    assertThatExceptionOfType(CommandLine.ParameterException.class)
+        .isThrownBy(() -> commandLine.parseArgs(args))
+        .withMessageStartingWith(
+            "Unable to read yaml configuration. Invalid yaml file [%s]", configFile);
+  }
+
+  @Test
+  void emptyYamlConfigFileThrowsExceptionDuringParsing(@TempDir final Path tempDir)
+      throws IOException {
+    final String extraYamlOptions = "";
+    final File configFile =
+        Files.writeString(tempDir.resolve("config.yaml"), extraYamlOptions).toFile();
+    final Eth2SignerCommand eth2SignerCommand = new Eth2SignerCommand();
+    final CommandLine commandLine = new CommandLine(eth2SignerCommand);
+    commandLine.registerConverter(Level.class, Level::valueOf);
+    commandLine.setDefaultValueProvider(new YamlConfigFileDefaultProvider(commandLine, configFile));
+
+    final String[] args = CmdlineHelpers.validBaseCommandOptions().split(" ");
+    assertThatExceptionOfType(CommandLine.ParameterException.class)
```

Comment:
Do we care if the config file is empty? Since we have default options for everything having an empty config could be considered valid?

Reply:
The check was there in Besu's "toml" configuration. This implementation is roughly based on that.